zebra-consensus/src/primitives/groth16.rs

```diff
@@ -68,6 +68,22 @@ pub type BatchVerifyingKey = VerifyingKey<Bls12>;
 /// This is the key used to verify individual items.
 pub type ItemVerifyingKey = PreparedVerifyingKey<Bls12>;
 
+lazy_static::lazy_static! {
+    /// The Sapling Groth16 verifying key for spends.
+    pub static ref SAPLING_SPEND_VERIFYING_KEY: std::sync::OnceLock<crate::groth16::VerifyingKey<Bls12>> = {
+        let cell = std::sync::OnceLock::new();
+        let _ = cell.set(GROTH16_PARAMETERS.sapling.spend.verifying_key().inner().clone());
+        cell
+    };
+
+     /// The Sapling Groth16 verifying key for outputs.
+    pub static ref SAPLING_OUTPUT_VERIFYING_KEY: std::sync::OnceLock<crate::groth16::VerifyingKey<Bls12>> = {
+        let cell = std::sync::OnceLock::new();
+        let _ = cell.set(GROTH16_PARAMETERS.sapling.output.verifying_key().inner().clone());
+        cell
+    };
+}
```

Comment:
I think `lazy_static` is the way to go here, the `OnceLock` seems unnecessary (and may add a bit of synchronization overhead):

```suggestion
lazy_static::lazy_static! {
    /// The Sapling Groth16 verifying key for spends.
    pub static ref SAPLING_SPEND_VERIFYING_KEY: crate::groth16::VerifyingKey<Bls12> =
        GROTH16_PARAMETERS.sapling.spend.verifying_key().inner().clone();

     /// The Sapling Groth16 verifying key for outputs.
    pub static ref SAPLING_OUTPUT_VERIFYING_KEY: crate::groth16::VerifyingKey<Bls12> =
        GROTH16_PARAMETERS.sapling.output.verifying_key().inner().clone();
}
```

`Box::leak()` would work too (though it would have to be repeated elsewhere, so lazy_static seems better):

```rust
let vk = Box::new(
    GROTH16_PARAMETERS
        .sapling
        .spend
        .verifying_key()
        .inner()
        .clone(),
);

let vk: &VerifyingKey<_> = Box::leak(vk);
```

Reply:
Thanks, it seems the `OnceLock` was left there from a different attempt i did before. Fixed in https://github.com/ZcashFoundation/zebra/pull/8527/commits/d7ad18976facdeee9fb533fdda8ec4aec56ad0fb